pkg/pgmodel/series_set.go

```diff
@@ -0,0 +1,198 @@
+package pgmodel
+
+import (
+	"fmt"
+	"math"
+	"sort"
+
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v4"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+)
+
+const (
+	// Postgres time zero is Sat Jan 01 00:00:00 2000 UTC.
+	// This is the offset of the Unix epoch in milliseconds from the Postgres zero.
+	PostgresUnixEpoch = -946684800000
+)
+
+var (
+	errInvalidData = fmt.Errorf("invalid row data")
+)
+
+// pgxSeriesSet implements storage.SeriesSet.
+type pgxSeriesSet struct {
+	rowIdx int
+	rows   []pgx.Rows
+	err    error
+}
+
+// Next forwards the internal cursor to next storage.Series
+func (p *pgxSeriesSet) Next() bool {
+	if p.rowIdx >= len(p.rows) {
+		return false
+	}
+	for !p.rows[p.rowIdx].Next() {
+		p.rows[p.rowIdx].Close()
+		p.rowIdx++
+		if p.rowIdx >= len(p.rows) {
+			return false
+		}
+	}
+	return true
+}
+
+// At returns the current storage.Series. It expects to get rows to contain
+// four arrays in binary format which it attempts to deserialize into specific types.
+// It also expects that the first two and second two arrays are the same length.
+func (p *pgxSeriesSet) At() storage.Series {
+	if p.rowIdx >= len(p.rows) {
+		return nil
+	}
+	arrays := p.rows[p.rowIdx].RawValues()
```

Comment:
This should be p.rows[p.rowIdx].Scan(&ps.labelNames, &ps.labelValues, &ps.times, &ps.values)